vignettes/wflow-06-gitlab.Rmd

```diff
@@ -0,0 +1,147 @@
+uses `sed` to do this but you could also do it in RStudio or your favourite text
+editor.
+
+```
+sed s/docs/public/ analysis/_site.yml
+```
+
+We already have a directory called `docs` so we also need to tell git that we
+want to change the name:
+
+```
+git mv docs/ public/
+```
+
+## Step 3: Add a GitLab CI config file
+
+While we can easily set up web page hosting for GitHub using just the website
+interface for GitLab we need to do a bit more work. GitLab using their
+continuous integration system to manage building and hosting website so
+we need to provide a config file with instructions on how to do this. This
+file should be called `.gitlab-ci.yml` and be placed at the root of our project
+directory. Because our HTML pages have already been built by workflowr we don't
+actually need GitLab to run any code for us so our config file can be quite
+simple. Here is an example of what the config file should look like:
+
+```
+pages:
+    stage: deploy
+    script:
+        - echo 'Nothing to do...'
+    artifacts:
+        paths:
+            - public
+    only:
+        - master
+```
+
+> **NOTE:** If you are using a private GitLab instance you might also need to
+set up a GitLab runner and assign it to your project. Instructions on how to
+do that are outside the scope of this vignette.
+
+At this stage we should have everything we need to host our project. Let's
+commit what we have and push it to GitLab.
+
+```
+git add .gitlab-ci.yml
+git add analysis/_site.yml
+git commit -m "Switch to hosting with GitLab"
+git push origin master
+```
+
+We should only have to do this manually the first time, after that the normal
+`wflow_publish` function should work.
+
+If this has worked correctly should now be able to see your GitLab hosted
+workflowr project website! To find the link to the website go to your GitLab
+repository page and select "Pages" under the "Settings" menu on the left. If
+the link is missing or doesn't work the job to build the website may not have
+run successfully. You can check this by clicking on the "CI / CD" menu.
+
+## Step 4: Fixing the links
+
+If you look carefully at your new website you should notice that many of the
+links that are usually there are missing. This is a result of us not using the
+`wflow_git_remote` command earlier. To fix these links we need to add the
+following line to the `_workflowr.yml` file.
+
+```
+github: GITLAB_REPOSITORY_URL
+```
+
+Here GITLAB_REPOSITORY_URL should be a URL to your project GitLab repository,
+not the URL for your project website.
+
+To fix the links in our exisiting web pages we need to rebuild them. We can
+force workflowr to do this by setting the `republish` option in the 
+`wflow_publish` function to `TRUE`.
+
+```
+wflow_publish(republish = TRUE)
+```
+
+Check your website to make sure you have links to previous versions of versions
+of code, figures etc. Note that the links to previous HTML documents on GitLab
+will not work with the current version of workflowr.
```

Comment:
I confirmed the links to the R Markdown files worked. I wasn't able to get the links to the figures to work, so I'll need to investigate that.

For hosting the HTML, I'll need to switch to raw.githack (which actually uses slightly different URLs for GitHub vs. GitLab) and also remove the appropriate string depending on the host URL ([line](https://github.com/jdblischak/workflowr/blob/ee111e2dbc730d7ec21dc62770eb963889e9fc3b/R/report.R#L122)).

Reply:
👍 I'm not sure if it will work on self hosted GitLab anyway but if you can send the correct link format happy to test.